File: src/Mobile_Customer/lib/features/home/domain/entities/voucher.dart
import 'package:intl/intl.dart';

class Voucher {
  final int id;
  final int storeId;
  final int? serviceId;

  final int discountType;
  final double discountValue;
  final double minOrderValue;
  final double maxDiscount;

  final String code;
  final String? imageUrl;

  final DateTime? startDate;
  final DateTime? endDate;

  final String? serviceName;
  final double originalPrice;
  final double discountedPrice;

  Voucher({
    required this.id,
    required this.storeId,
    this.serviceId,
    required this.discountType,
    required this.discountValue,
    required this.minOrderValue,
    required this.maxDiscount,
    required this.code,
    this.imageUrl,
    this.startDate,
    this.endDate,
    this.serviceName,
    this.originalPrice = 0,
    this.discountedPrice = 0,
  });

  String get discountText {
    if (discountType == 0) {
      final perc =
          discountValue % 1 == 0 ? discountValue.toInt() : discountValue;
      return '${perc}%';
    } else {
      final f =
          NumberFormat.currency(locale: 'vi_VN', symbol: '', decimalDigits: 0);
      return '${f.format(discountValue)}₫';
    }
  }
}

File: src/Backend_API/BeautyBookingSystem.Domain/Entities/Voucher.cs
using BeautyBookingSystem.Domain.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BeautyBookingSystem.Domain.Enums;

namespace BeautyBookingSystem.Domain.Entities
{
    public class Voucher : BaseEntity
    {
        public int StoreId { get; set; }
        public virtual Store Store { get; set; } = null!;

        public string Code { get; set; } = string.Empty;
        public DiscountType DiscountType { get; set; }
        public decimal DiscountValue { get; set; }
        public decimal MinOrderValue { get; set; }
        public decimal MaxDiscount { get; set; }

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int UsageLimit { get; set; }
        public int UsedCount { get; set; }
        public string? ImageUrl { get; set; }

        public int? ServiceId { get; set; }
        public virtual Service? Service { get; set; }

        public virtual ICollection<UserVoucher> UserVouchers { get; set; } = new List<UserVoucher>();
        public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}
